recipes/log4cxx/all/conanfile.py

```diff
@@ -12,138 +15,143 @@ class Log4cxxConan(ConanFile):
     url = "https://github.com/conan-io/conan-center-index"
     license = "Apache-2.0"
     homepage = "https://logging.apache.org/log4cxx"
-    topics = ("logging", "log")
+    topics = ("logging")
 
     settings = "os", "arch", "compiler", "build_type"
     options = {
         "shared": [True, False],
         "fPIC": [True, False],
+        "with_networking": [True, False],
+        "with_wchar_t": [True, False],
+        "with_fmt_layout": [True, False],
+        "with_odbc_appender": [True, False],
+        "with_multiprocess_rolling_file_appender": [True, False],
+        "with_smtp_appender": [True, False],
+        "with_qt": [True, False],
     }
     default_options = {
         "shared": False,
         "fPIC": True,
+        "with_networking": True,
+        "with_wchar_t": False,
+        "with_fmt_layout": False,
+        "with_odbc_appender": False,
+        "with_multiprocess_rolling_file_appender": False,
+        "with_smtp_appender": False,
+        "with_qt": False,
     }
 
-    generators = "cmake", "cmake_find_package", "pkg_config"
-    _cmake = None
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
-    @property
-    def _build_subfolder(self):
-        return "build_subfolder"
-
     def export_sources(self):
-        self.copy("CMakeLists.txt")
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            self.copy(patch["patch_file"])
+        export_conandata_patches(self)
 
     def config_options(self):
         if self.settings.os == "Windows":
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
+            self.options.rm_safe("with_smtp_appender") # *nix only
+        elif Version(self.version) <= "1.0.0": # SMTP appender is broken in version 0.12 through 1.0.0
+            self.options.rm_safe("with_smtp_appender")
+        if Version(self.version) < "1.0.0":
+            self.options.rm_safe("with_multiprocess_rolling_file_appender")
+            self.options.rm_safe("with_networking")
+            self.options.rm_safe("with_fmt_layout")
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
 
     def requirements(self):
         self.requires("apr/1.7.0")
-        self.requires("apr-util/1.6.1")
-        self.requires("expat/2.4.2")
-        if self.settings.os != "Windows":
-            self.requires("odbc/2.3.9")
+        self.requires("apr-util/[>=1.6]")
```

Comment:
Please, avoid using version ranges for packages that are not allowed yet: https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/dependencies.md#version-ranges

Using version range does not guarantee compatibility between versions, it depends on each project. We are preparing a list of new packages allowed in CCI, but so far only `openssl` and `cmake` are allowed.